### ISSUE TITLE
Using Restructured Text?

### ISSUE THREAD

**Issue by dantleech** [inviqa/engineering-guidelines#6]
Markdown is accessible, but I don't think it's always the best solution, in particular I miss:

- Admonitions (`NOTE`, `INFO`, etc).
- Inter-document references

[Restructured text](http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) provides both of those and more, and the markup is very similar to markdown, e.g.

```
Header 1
=======

- Bullet one
- Bullet two

*italics* 
**bold**

Header 2
------------

Header 3
~~~~~~~
```

In addition you can automatically publish on readthedocs.org, theme it and convert to PDF, etc. Read the docs also provides search.

What do you think?

**Comment by peteward** [comment]
interesting idea, what do others think?

**Comment by dantleech** [comment]
> I presume it renders inline in github?

Yeah, but not 100%, this is the [symfony DI doc on github](https://github.com/symfony/symfony-docs/blob/master/components/dependency_injection.rst) and on [readthedocs](https://symfony.com/doc/current/components/dependency_injection.html) (I _think_ it's readthedocs).

**Comment by chicgeek** [comment]
I have a preference for markdown because I'm wary of creating barriers to contribution, especially at this early stage.

From what I understand, using restructured text would require:

* people to familiarise themselves with RST syntax (minor)
* a compilation step when adding content with RST features

This compilation step also introduces the requirement to pull down the repo while now we can be quite lazy and edit / preview files directly on GitHub. Anything to make contributing easier is a boon.

This is not to preclude us from converting the project at a later time if we identify the benefits of these features outweigh the cost of maintenance. Is this something we can defer decision on until we have a bit more content to contend with?

**Comment by peteward** [comment]
that sounds wise. 

I think ease of contribution trumps anything else right now - I didn't realise there would be a compilation step.

I'm also not sure that we need to publish these anywhere else. Maybe we will have a desire to do so in future.

**Comment by richardmiller** [comment]
Yeah, rst works well for things like Symfony docs where automatic content and index generation is useful. If we get to the point where we need those then we probably have too much content and should be looking to solve that directly.

That's not to say there are not other useful things like notes, codeblocks etc. that we may want to consider in the future.

**Comment by dantleech** [comment]
> This compilation step also introduces the requirement to pull down the repo while now we can be quite lazy and edit / preview files directly on GitHub.

This isn't really true - we can still edit on GH and when the change is saved, the docs will recompile automatically, and GH still renders RST pretty good.

For the biggest advantage is integrity, you will not suffer from broken links with RST.

Saying that, it is defintely beneficial for non-minor edits to compile locally (faster feedback cycle) and I your're right, that is a barrier to entry we don't really want.

And there are other ways to solve this I'm sure.
